src/reports/data/sonata_data.hpp

```diff
@@ -6,9 +6,35 @@
 #include "node.hpp"
 
 class SonataData {
+    using nodes_t = std::map<uint64_t, Node>;
+  public:
+    SonataData(const std::string& report_name, size_t max_buffer_size, int num_steps, double dt, double tstart, double tend, std::shared_ptr<nodes_t> nodes);
+    SonataData(const std::string& report_name, const std::vector<double>& spike_timestamps, const std::vector<int>& spike_node_ids);
+
+    void prepare_dataset();
+    void write_report_header();
+    void write_spikes_header();
+    void write_data();
+    void close();
+
+    bool is_due_to_report(double step);
```

Comment:
Shouldn't this member method be `const`?

Reply:
indeed